src/pip/_internal/commands/download.py

```diff
@@ -141,10 +141,12 @@ def run(self, options, args):
             options.abi,
             options.implementation,
         ])
-        binary_only = FormatControl(set(), {':all:'})
+        fmt_ctl = FormatControl(set(), {':all:'})
         no_sdist_dependencies = (
-            options.format_control != binary_only and
-            not options.ignore_dependencies
+            (
+                options.format_control.no_binary != fmt_ctl.no_binary or
```

Comment:
It might make sense to add equality checks to the FormatControl class. This would allow this comparison to stay as is.